controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -193,6 +218,43 @@ func removePath(patchMap map[string]interface{}, path contract.Path) {
 	}
 }
 
+// enforcePath a path from simpleMap into the twoWayMap thus
+// enforcing changes aligned to the modified object for the authoritative paths.
+func enforcePath(simpleMap, twoWayMap map[string]interface{}, path contract.Path) {
+	switch len(path) {
+	case 0:
+		// If path is empty, no-op.
+		return
+	case 1:
+		// If we are at the end of a path, enforce the value.
+		twoWayMap[path[0]] = simpleMap[path[0]]
+	default:
+		// If in the middle of a path, go into the nested map,
+		nestedSimpleMap, ok := simpleMap[path[0]].(map[string]interface{})
+
+		// If the value is not a map (is a value or a list), return (not a full match).
```

Comment:
No, because it is a partial match.
Assume we want to be authoritative on `metadata.lables`, but this in the target object there is only `metadata` with no `labels` but something else, e.g. `finalizers`. 

This check prevents us to enforce the entire `metadata` struct that contains `finalizers`, that should not be autoritative.

Reply:
@fabriziopandini I wonder if it would be better to make metadata.labels and metadata.annotations explicitly authoritative instead of relying on this check that we recursively only make maps authoritative

Also, based on the code I would have assumed that we hit `case 1` where `path[0]` is `metadata` and then just set the whole map.

Reply:
In this case it is the same given that patch helper consider changes only in metadata.labels, metadata.annotation and spec (allowed paths).
but I have made this more explicit

Reply:
Good point, thx